staging/src/k8s.io/kubectl/pkg/cmd/plugin/plugin_completion.go

```diff
@@ -70,19 +81,20 @@ func SetupPluginCompletion(cmd *cobra.Command, args []string) {
 			// to avoid them being included in the completion choices.
 			// This must be done *before* adding the plugin commands so that
 			// when creating those plugin commands, the flags don't exist.
-			cmd.Root().ResetFlags()
+			kubectl.ResetFlags()
 			cobra.CompDebugln("Cleared global flags for plugin completion", true)
 
-			addPluginCommands(cmd)
+			registerPluginCommands(kubectl, math.MaxInt, true)
 		}
 	}
 }
 
-// addPluginCommand adds a Cobra command to the command tree
-// for each plugin so that the completion logic knows about the plugins
-func addPluginCommands(cmd *cobra.Command) {
-	kubectl := cmd.Root()
-	streams := genericiooptions.IOStreams{
+// registerPluginCommand allows adding Cobra command to the command tree or extracting them for usage in
+// e.g. the help function or for registering the completion function
+func registerPluginCommands(kubectl *cobra.Command, depth int, register bool) (cmds []*cobra.Command) {
```

Comment:
Nit: rather than having 2 arguments `depth` and `register`, how about we just add `list bool` and if that is set it means we only care about the highest level (ie. yours `depth = 1` and `register = false`), when false, we register like in the old days. I don't think we need that much control over the process.

Reply:
Just did! Lmk if there's anything else please